lib/Arch/Runtime/HyperCall.cpp

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright (c) 2022 Trail of Bits, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+Memory *__remill_sync_hyper_call(State &state, Memory *mem,
+                                 SyncHyperCall::Name call) {
+  switch (call) {
+
+#if defined(REMILL_ON_X86) or defined(REMILL_ON_AMD64)
+
+    case SyncHyperCall::kX86CPUID:
+      asm volatile("cpuid"
+                   : "=a"(state.gpr.rax.aword), "=b"(state.gpr.rbx.aword),
+                     "=c"(state.gpr.rcx.aword), "=d"(state.gpr.rdx.aword)
+                   : "a"(state.gpr.rax.aword), "b"(state.gpr.rbx.aword),
+                     "c"(state.gpr.rcx.aword), "d"(state.gpr.rdx.aword));
+      break;
+
+    case SyncHyperCall::kX86ReadTSC:
+      asm volatile("rdtsc"
+                   : "=a"(state.gpr.rax.dword), "=d"(state.gpr.rdx.dword));
+      break;
+
+    case SyncHyperCall::kX86ReadTSCP:
+      asm volatile("rdtscp"
+                   : "=a"(state.gpr.rax.aword), "=c"(state.gpr.rcx.aword),
+                     "=d"(state.gpr.rdx.aword)
+                   : "a"(state.gpr.rax.aword), "c"(state.gpr.rcx.aword),
+                     "d"(state.gpr.rdx.aword));
+      break;
+
+    case SyncHyperCall::kX86LoadGlobalDescriptorTable:
+      // TODO(alex): Where do I get the operand from?
```

Comment:
And this.

Reply:
https://github.com/lifting-bits/remill/blob/f096390b95d8528ff28e7e8d178b3577e34e0caa/lib/Arch/X86/Semantics/SYSTEM.cpp#L19-L62

`state.addr_to_load`.

Reply:
[e58a2f2](https://github.com/lifting-bits/remill/pull/611/commits/e58a2f291cac43263b864e32bee709fe59a4dc35)